### PR TITLE
[Auto Parallel] add dropout spmd rules

### PR DESCRIPTION
**Pull request by deepllz**
### PR Category

Auto Parallel

### PR Types

Performance

### Description

dropout切分推导规则添加
Pcard-73145
dropout切分推导会导致dp情况下精度无法诸位对齐，gpt中使用了dropout，因此gpt3个开启dp的单测会无法通过。暂时先禁止这3个测试，待此PR合入之后，再修改paddleNLP中gpt的测试。

### CODE REVIEW

File: paddle/phi/infermeta/spmd_rules/dropout.cc [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+#include "paddle/phi/infermeta/spmd_rules/elementwise.h"
+#include "paddle/phi/infermeta/spmd_rules/spmd_rule_macro_define.h"
+#include "paddle/phi/infermeta/spmd_rules/utils.h"
+
+namespace phi {
+namespace distributed {
+
+// args : (Tensor x, Tensor seed_tensor, Scalar p, bool is_test, str mode, int
+// seed, bool fix_seed) output : Tensor(out), Tensor(mask)
+SpmdInfo DropoutFwdInferSpmd(const DistMetaTensor& x,
```
Comment (jeff41404) [inline paddle/phi/infermeta/spmd_rules/dropout.cc:30]: In general, the naming of forward spmd rule is `DropoutInferSpmd`

File: paddle/phi/infermeta/spmd_rules/dropout.cc [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+      common::errors::InvalidArgument("seed_tensor should be None"));
+  SpmdInfo info = ElementwiseUnaryInferSpmd(x);
+  info.first.push_back(seed_tensor.dist_attr());
+  info.second.push_back(info.second[0]);
+  return info;
+}
+
+// args : (Tensor mask, Tensor out_grad, Scalar p, bool is_test, str mode)
+// output : Tensor(x_grad)
+SpmdInfo DropoutBwdInferSpmd(const DistMetaTensor& mask,
```
Comment (jeff41404) [inline paddle/phi/infermeta/spmd_rules/dropout.cc:52]: In general, the naming of backward spmd rule is `DropoutGradInferSpmd`

File: paddle/phi/infermeta/spmd_rules/dropout.cc [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+                             const DistMetaTensor& out_grad,
+                             Scalar p,
+                             bool is_test,
+                             const std::string& mode) {
+  return ElementwiseBinaryInferSpmd(mask, out_grad);
+}
```
Comment (jeff41404) [inline paddle/phi/infermeta/spmd_rules/dropout.cc:58]: shall we add `DropoutInferSpmdReverse` to support reverse spmd rule in forward when use static mode

Comment (deepllz) [reply paddle/phi/infermeta/spmd_rules/dropout.cc:58]: 现在不需要添加reverse的切分推导了

**Review by zhiqiu — APPROVED** [review]
LGTM

**Review by zyfncg — APPROVED** [review]

**Review by liym27 — APPROVED** [review]
LGTM

File: paddle/phi/infermeta/spmd_rules/dropout.cc (hunk 1 above)
Comment (deepllz) [reply paddle/phi/infermeta/spmd_rules/dropout.cc:30]: 由于现在不需要写reverse，所以命名为fwd和bwd更合理，已经有部分spmd是如此命名的了

File: paddle/phi/infermeta/spmd_rules/dropout.cc (hunk 2 above)
Comment (deepllz) [reply paddle/phi/infermeta/spmd_rules/dropout.cc:52]: 同上

**Pull request merged**